### ISSUE TITLE
Better build instructions on all platforms

### ISSUE THREAD

**Issue by justinfarrelldev** [civctp2/civctp2#398]
Checking out both #379 and #348, as well as my own issue #384, it seems apparent that very few people can actually build this repo. The makefile goes a long way as well as the Autogen and Configure files, but overall the documentation on how to build the games is sparse and ancient. This is leading to a lot of people being unable to contribute to the development of the Apolyton Edition who would otherwise be happy to - myself and several of the people on the Discord server have no idea how to build this edition, and there isn't even any releases to download. Can we please get revised instructions on how to build the Apolyton Edition of the game for multiple platforms (Windows and Linux at least) that are specific, clear and easy to follow? Specifically, I think the instructions should mention: 

- the specific compiler which was used to build the game 
- the specific OS used for the instructions
- how to generate the required .deb files 
- the specific IDE and version used (if relevant)

Basically, just a from-scratch list of instructions that list everything done to get the Apolyton Edition working to a point where one can start contributing. 

Thank you.

**Comment by myocytebd** [comment]
I think it should be not be a big problem to build the repo on linux.  
- The only building system problem that I ran into (and rather obvious) is that compiling & .cdb generator does not sync.  
So it is often necessary to build twice if .cdb is modified - 1st pass fails due to stale .cdb gen; 2nd pass succeeds because .cdb generator does not regen and thus free of race condition.  
- The source code quality is another story, and may leads to other building problems.  

This is the flags I used to compile the repo on my local machine: Ubuntu 20.04 / gcc 9.4  
Please notes: 
* Most ccflags are to suppress less significant and noisy warnings.  
* Some ccflags are necessary because of my local changes.  
```
SCRIPT_DIR="$(dirname "$(readlink -f "$0")")"

if [[ "$1" == "--d" ]] || [[ "$1" == "--debug" ]]; then
    export CFLAGS="$CFLAGS -Og"
    export CXXFLAGS="$CXXFLAGS -Og"
else
    export CFLAGS="$CFLAGS -O2"
    export CXXFLAGS="$CXXFLAGS -O2"
fi

export CFLAGS="$CFLAGS -g -fuse-ld=gold -Wno-unused-function"
export CXXFLAGS="$CXXFLAGS -g -fno-omit-frame-pointer -fno-optimize-sibling-calls -fuse-ld=gold -std=gnu++17" # -fno-exceptions # FIXME: enable -fno-exceptions

# Fatal
# gcc 9.4 may generate broken code if return statement is missing.
export CXXFLAGS="$CXXFLAGS -Werror=return-type"

# Pointless / Too noisy
export CXXFLAGS="$CXXFLAGS -Wno-maybe-uninitialized"
# Probably safe
export CXXFLAGS="$CXXFLAGS -Wno-write-strings -Wno-reorder -Wno-unused-function -Wno-endif-labels -Wno-multichar -Wno-char-subscripts -Wno-conversion-null -Wno-register"
# Probably safe, but may miss true errors
export CXXFLAGS="$CXXFLAGS -Wno-unused-but-set-variable -Wno-unused-value -Wno-switch -Wno-misleading-indentation -Wno-parentheses -Wno-vla -Wno-unknown-pragmas"
# Unsafe, but probably insignificant
export CXXFLAGS="$CXXFLAGS -Wno-format -Wno-format-overflow"
export CXXFLAGS="$CXXFLAGS -Wno-strict-aliasing"
# Unsafe
export CXXFLAGS="$CXXFLAGS -Wno-class-memaccess -Wno-sign-compare"
# Error
export CXXFLAGS="$CXXFLAGS -fpermissive -Wno-address-of-packed-member -Wno-deprecated-declarations"
# Critical Error
export CXXFLAGS="$CXXFLAGS -Wno-restrict"

export LIBS="-lunwind"
$SCRIPT_DIR/configure --enable-silent-rules
```

**Comment by myocytebd** [comment]
I feel that this repo is dead.  
If you could find a group of people contributing to it, IMHO you had better fork it.  
esp. because GitHub PR workflow is horrible that it can easily cost more effort to submit patches than writing patches.  
Contributing non-trivial patches could be nightmare for non-commiters.

**Comment by justinfarrelldev** [comment]
> I feel that this repo is dead. If you could find a group of people contributing to it, IMHO you had better fork it. esp. because GitHub PR workflow is horrible that it can easily cost more effort to submit patches than writing patches. Contributing non-trivial patches could be nightmare for non-commiters.

Thank you for the detailed instructions! I will add a link to this on the Discord for people trying to build it. 

Also, I have been trying to get this running from VS 2022 the past two days and have gotten fairly far. Build was successful, but went to run it and got this popup. I'll update this if I find anything to fix it.
![image](https://user-images.githubusercontent.com/42923203/219923492-dc356b85-56d8-4bbe-9e27-bd8924f80a6c.png)

**Comment by MartinGuehmann** [comment]
> Also, I have been trying to get this running from VS 2022 the past two days and have gotten fairly far. Build was successful, but went to run it and got this popup. I'll update this if I find anything to fix it.

No, you did not run, at least that what I guess, since the message box is covering the interesting information about ctp2.exe. I guess the creation is is 12/26/2022 9:09 PM and its size is about 6 MB, which is the release version. The version you compiled this year should be well from this year and its size should be around 4 MB.

I guess you did not copy it over.

The released version of the executable is in ctp2_program/ctp, while the freshly compiled version is in ctp2_code/ctp. Depending on the configuration you chose, the file will be named differently. Configuration Debug generates CivCTP_dbg.exe, the heavy debug version with logging, assertions, and memory leaks checking. The last thing makes it slow.

Then the configuration Release generates ctp2r.exe, so obviously not the version that was actually released. I think it contains some minor stuff for debugging than the final version.

And that what you get with the configuration Final, that it's then ctp2.exe.

Of course, Visual Studio is an integrated development environment. That means you start the game within Visual Studio. You can just press F5, that should start also the debugger or CTRL+F5 that should start it without the debugger. You can check that in the Debug menu, right now I am on a Linux so I can't check it and that is already quite some unconscious routine for me.

If you want to use your save games, then copy the folder ctp2_program/ctp/save/ including its subfolders to ctp2_code/ctp/save/.

**Comment by LynxAbraxas** [comment]
> I feel that this repo is dead. 

@myocytebd What makes you feel that?

**Comment by rambo919** [comment]
A clear warning is needed in the instructions. "configure: error: unsafe absolute working directory name" is too vague, not everyone will realize that no spaces in the path is assumed.

**Comment by rambo919** [comment]
> > I feel that this repo is dead.
> 
> @myocytebd What makes you feel that?

Nothing to do with feelings just his bad way of stating it. Most files have no recent edits, lots of open issues, latest edits are a year old, lots of open issues directly related to people being unable to build.

**Comment by LynxAbraxas** [comment]
> > > I feel that this repo is dead.
> > 
> > @myocytebd What makes you feel that?
> 
> Nothing to do with feelings just his bad way of stating it. Most files have no recent edits, lots of open issues, latest edits are a year old, lots of open issues directly related to people being unable to build.

Contributions for improvement are very welcome!
Most of the long term contributors and maintainers have not so much time any more;-)

**Comment by MartinGuehmann** [comment]
I merged in the updated build instructions into the readme. So if you still have an issue please open a new one.

**Comment by LynxAbraxas** [comment]
Please, also let us know if things work out for you now and what caused the problems.

**Comment by justinfarrelldev** [comment]
> Please, also let us know if things work out for you now and what caused the problems.

I will definitely try building it again in a few weeks when my new computer comes in. Thank you!
